src/workflow/ArcanistPatchWorkflow.php

```diff
@@ -513,16 +513,22 @@ public function run() {
           pht('Base commit is not in local repository; trying to fetch.'));
         // UBER CODE
         $this->authenticateConduit();
-        $this->pullFromAllRemotesUntilFound($bundle->getBaseRevision());
-        $has_base_revision = $repository_api->hasLocalCommit(
-          $bundle->getBaseRevision());
+        if ($source == self::SOURCE_DIFF) {
+          $this->pullBaseTagFromStagingArea($param);
+          $has_base_revision = $repository_api->hasLocalCommit(
+            $bundle->getBaseRevision());
+        }
```

Comment:
It looks duplicate of line 437

Reply:
Not entirely - to pull tag/ref there you also need --use-uber-staging-tags, but if that argument is passed - then it is identical code. I'll be removing that functionality later once we're happy with behavior/performance.